### ISSUE TITLE
WCAG 2.1 AA compliance

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#4828]
This is an evolving periodically updated (_last updated 9/1/2020_) list that represents our current understanding of the overall accessibility of Carbon Components React across four spectrums:

- 🤖 **has automated accessibility testing issues/violations**
  - we work toward compliance with _all_ known automated testing systems, but use DAP/Accessibility Checker (IBM's in house tools) for compliance tracking.
- ⌨️ **has keyboard accessibility issues/violations** 
  - Any and all mouse interactions are expected to be operable using only the keyboard. And all interactable elements should be reachable in a logical way/order. We defer to the WAI-ARIA Authoring Practices Guidelines.
- 👁️‍🗨️ **has visual accessibility issues/violations**
  - All colors/themes (etc) should meet or improve on WCAG 1.2 guidelines for color contrast. Wayfinding -- or focus states -- should be clear and perceivable in all themes and colors as well
- 👂 **has screen reader accessibility issues/violations**
  - All interactions and content should be accessible using JAWS (in Chrome/Edge and Firefox on Windows 10), NVDA (in Chrome/Edge and Firefox on Windows 10), and VoiceOver (on Safari in macOS)

---

1. Accordion
2. Breadcrumb
3. Button👁️‍🗨️
4. Checkbox👁️‍🗨️⌨️
5. Code snippet 👂 
6. Content switcher👁️‍🗨️⌨️
7. Data table 👁️‍🗨️👂 
8. Date picker⌨️👁️‍🗨️
9. Dropdown🤖⌨️ 👂 
10. File uploader
11. Form
12. Inline loading⌨️👂 
13. Link
14. List
15. Loading
16. Modal👁️‍🗨️👂 
17. Notification
18. Number input👂
19. Overflow menu👁️‍🗨️⌨️
20. Pagination👁️‍🗨️
21. Progress indicator
22. Radio button👁️‍🗨️
23. Search
24. Select👁️‍🗨️
25. Slider
26. Structured list⌨️👂
27. Tabs👂
28. Tag
29. Text input
30. Tile
31. Toggle
32. Tooltip👂
33. UI Shell👂

**Comment by pascal-potvin** [comment]
Just wanted to see if it was possible to get an updated timeline for the ones which have  🚨open accessibility issues. Our products rely heavily on accessibility. Thanks

**Comment by dakahn** [comment]
We're close! Our target is by the end of Spring. We were waylayed for a few cycles while we upgraded a dependency for our dropdown components but with that work done we're cracking into the remaning 8 with known violations. ✌️

**Comment by mattrosno** [comment]
@dakahn I didn't mean to close this epic if it's still getting used?

**Comment by dakahn** [comment]
Yeah! Still using it to broadly track our closed vs open issues. I think it's still helpful.
